workflow_scripts/test_models.py

```diff
@@ -1,48 +1,80 @@
-stdoutput, stderroutput = obtain_diff.communicate()
-diff_list = stdoutput.split()
-
-# identify list of changed onnx models in model Zoo
-model_list = [str(model).replace("b'","").replace("'", "") for model in diff_list if ".onnx" in str(model)]
-
-# run lfs install before starting the tests
-run_lfs_install()
-
-print("\n=== Running ONNX Checker on added models ===\n")
-# run checker on each model
-failed_models = []
-for model_path in model_list:
-    model_name = model_path.split('/')[-1]
-    print("Testing:", model_name)
-
-    try:
-      pull_lfs_file(model_path)
-      model = onnx.load(model_path)
-      onnx.checker.check_model(model)
-      print("Model", model_name, "has been successfully checked!")
-    except Exception as e:
-      print(e)
-      failed_models.append(model_path)
-
-if len(failed_models) != 0:
-  print(str(len(failed_models)) +" models failed onnx checker.")
-  sys.exit(1)
-
-print(len(model_list), "model(s) checked.")
+
+def main(argv):
+  # default test by onnx and onnxruntime
+  test_onnx, test_onnxruntime = True, True
+  # test by onnx only
+  if len(argv) >= 2 and argv[1] == 'onnx':
+    test_onnxruntime = False
+  # test by onnxruntime only
+  elif len(argv) >= 2 and argv[1] == 'onnxruntime':
+    test_onnx = False
+    
+  cwd_path = Path.cwd()
+  # obtain list of added or modified files in this PR
+  obtain_diff = subprocess.Popen(['git', 'diff', '--name-only', '--diff-filter=AM', 'origin/master', 'HEAD'],
+  cwd=cwd_path, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  stdoutput, stderroutput = obtain_diff.communicate()
+  diff_list = stdoutput.split()
+
+  # identify list of changed onnx models in model Zoo
+  model_list = [str(model).replace("b'","").replace("'", "") for model in diff_list if ".onnx" in str(model)]
+  # run lfs install before starting the tests
+  test_utils.run_lfs_install()
+
+  print('\n=== Running ONNX Checker on added models ===\n')
+  # run checker on each model
+  failed_models = []
+  tar_ext_name = '.tar.gz'
+  for model_path in model_list:
+      model_name = model_path.split('/')[-1]
+      tar_name = model_name.replace('.onnx', tar_ext_name)
+      print('==============Testing {}=============='.format(model_name))
+
+      try:
+        # replace '.onnx' with '.tar.gz'
+        tar_gz_path = model_path[:-5] + '.tar.gz'
+        test_data_set = []
+        # if tar.gz exists, use the onnx model
+        if os.path.exists(tar_gz_path):
+          test_utils.pull_lfs_file(tar_gz_path)
+          # check whether 'test_data_set_0' exists
+          model_path_from_tar, test_data_set = test_utils.extract_test_data(tar_gz_path)
+
+          # 1. check the onnx model from .tar.gz
+          if test_onnx: check_model.by_onnx(model_path_from_tar, tar_name)
+          if test_onnxruntime: check_model.by_onnxruntime(model_path_from_tar, tar_name, test_data_set)
+        else:
+          test_utils.pull_lfs_file(model_path)
```

Comment:
should this be under else? Do you need this else? If onnx model is not downloaded then the code on line 57 won't run right?

Reply:
Thank you for catching this bug. You are right.